pandas/core/dtypes/dtypes.py

```diff
@@ -2102,8 +2102,9 @@ def type(self):
         elif pa.types.is_null(pa_type):
             # TODO: None? pd.NA? pa.null?
             return type(pa_type)
-        else:
-            raise NotImplementedError(pa_type)
+        elif isinstance(pa_type, pa.ExtensionType):
+            return type(self)(pa_type.storage_type).type
```

Comment:
is this going to match the storage_type?

Reply:
Yeah, the python type of the `storage_type`